src/main/java/dk/kb/discover/SolrService.java

```diff
@@ -411,44 +412,57 @@ private SolrParamMerger createBaseParams(
      * @return an URI ready for use with a HTTP component.
      */
     private URI createRequest(String handler, SolrParamMerger params) {
-        UriBuilder builder = UriBuilder.fromUri(server)
-                .path(path)
-                .path(solrCollection)
-                .path(handler);
-        params.forEach((key, values) ->
-                values.forEach(value ->
-                        builder.queryParam(key, value)));
-        return builder.build();
+        try {
+            URIBuilder builder = new URIBuilder(server)
+                    .setPathSegments(path, solrCollection, handler);
+            params.forEach((key, values) ->
+                    values.forEach(value ->
+                            builder.addParameter(key, value)));
+            return builder.build();
+        } catch (URISyntaxException e) {
+            log.error("Unable to construct URL for Solr select call to server='{}', collection='{}' with parameters {}",
+                    server, solrCollection, params, e);
+            throw new InternalServiceException("Unable to construct URL for Solr request");
+        }
     }
 
-
     /**
      * Creates a Solr oriented URI builder specific for suggest
      * @return a pre-filled builder ready to be extended with caller specific parameters.
      */
-    private UriBuilder createSuggestRequestBuilder(String suggestDictionary,String suggestQuery, Integer suggestCount, String wt) {
-        return UriBuilder.fromUri(server)
-                .path(path)
-                .path(solrCollection)
-                .path(SUGGEST)
-                .queryParam(SUGGEST_Q, suggestQuery)
-                .queryParam(SUGGEST_DICTIONARY, suggestDictionary)
-                .queryParam(SUGGEST_COUNT, suggestCount)
-                .queryParam(WT, WT_ENUM.safeParse(wt));
+    private URI createSuggestRequestBuilder(String suggestDictionary, String suggestQuery, Integer suggestCount, String wt) {
+        try {
+            return new URIBuilder(server)
+                    .setPathSegments(path, solrCollection, SUGGEST)
+                    .addParameter(SUGGEST_Q, suggestQuery)
+                    .addParameter(SUGGEST_DICTIONARY, suggestDictionary)
+                    .addParameter(SUGGEST_COUNT, suggestCount.toString())
+                    .addParameter(WT, WT_ENUM.safeParse(wt).toString())
+                    .build();
+        } catch (URISyntaxException e) {
+            log.error("Unable to construct URL for Solr suggest call to server='{}', collection='{}' with " +
+                            "dict='{}', query='{}', count={}, wt='{}'",
+                    server, solrCollection, suggestDictionary, suggestQuery, suggestCount, wt, e);
+            throw new InternalServiceException(
+                    "Unable to construct URL for Solr suggest call for '" + suggestQuery + "'");
```

Comment:
These URI builder libs all look quite alike. I like the `add`-part of method names in this one. Nice addition with the error handling.